File: resume-search-portal-service/src/queues/queue.manager.ts
import amqp from "amqplib";
import { getEnvValue } from "../utils/env.utils";
import { resumeLogger } from "../libs/common.logger.libs";

class AMQPManager {
  private amqpUrl: string = getEnvValue("AMQP_URL") as string;
  private amqpPort: number = Number(getEnvValue("AMQP_PORT")) as number;
  private connection: amqp.ChannelModel | null = null;
  private channel: amqp.Channel | null = null;

  private async initalizeConnection() {
    try {
      if (!this.connection) {
        this.connection = await amqp.connect(this.amqpUrl);
        this.channel = await this.connection.createChannel();
        resumeLogger.info(`The RabbitMQ Connection Has Establish Successfullyo on port : ${this.amqpPort}`);
      }
    } catch (err: any) {
      resumeLogger.error(
        `Error Connecting to the RabbitMQ, Please The Service RabbitMQ is Running or not`
      );
      throw err;
    }
  }

  public async getChannel() {
    return new Promise(async (resolve, reject) => {
      if (!this.channel) {
        await this.initalizeConnection();
      }
      resolve(this.channel);
    });
  }

  public async closeChannel() {
    return new Promise(async (resolve, reject) => {
      if (this.channel) {
        await this.channel.close();
        this.channel = null;
      }
      resolve(true);
    });
  }
}

const getAMQPInstances = (): AMQPManager => {
  return new AMQPManager();
};

export default getAMQPInstances;

File: resume-analyzer-service/Queues/QueueManager.py
import pika

from Utils.EnvUtils import get_env_value



class QueueManager:
    connection = None
    channel = None
    def __init__(self):
        self.rabbitmq_host = get_env_value('AMQP_URL')
        self.rabbitmq_port = int(get_env_value('AMQP_PORT'))
        self.rabbitmq_heartbeat = int(get_env_value('AMQP_HEARTBEAT'))

    def initialize_connection(self):
        try:
            if self.connection is None and self.channel is None:
                self.connection = pika.BlockingConnection(pika.ConnectionParameters(host=self.rabbitmq_host,port=self.rabbitmq_port,heartbeat=self.rabbitmq_heartbeat))
                self.channel = self.connection.channel()
        except Exception as error:
            print(f'Error Initializing the RabbitMQ Connection, Due To {error}')


    def get_channel(self):
        if self.channel is None:
            self.initialize_connection()
            return self.channel

    def close_channel(self):
        if self.channel:
            self.channel.close()
            self.channel = None
            return True


def get_queue_manager_instances():
    return QueueManager()
